reward_score/readme_gen_pred.py

```
@@ -0,0 +1,111 @@
+import difflib
+from utils import register_reward_score, get_llm_response
+from concurrent.futures import ThreadPoolExecutor, TimeoutError
+
+# 单例线程池（可以在模块全局初始化一次）
+executor = ThreadPoolExecutor(max_workers=4)  # 根据你的并发情况调节
+
+def safe_similarity(solution_str, ground_truth, timeout=2.0, default_reward=0.0):
+    """
+    安全计算相似度：支持timeout & 异常捕获
```

Ruff (0.12.2)

Docstring contains ambiguous `：` (FULLWIDTH COLON). Did you mean `:` (COLON)?

(RUF002)